hash_reader.go

```diff
@@ -33,98 +34,156 @@ type HashReader interface {
 }
 
 type hashReader struct {
-	ionReader      ion.Reader
-	hasherProvider IonHasherProvider
+	ionReader ion.Reader
+	hasher    hasher
+	typeIon   ion.Type
 }
 
 func NewHashReader(ionReader ion.Reader, hasherProvider IonHasherProvider) HashReader {
-	hashReader := &hashReader{ionReader, hasherProvider}
+	hashReader := &hashReader{ionReader, *newHasher(hasherProvider), ion.NoType}
```

Comment:
We don't need to specify the typeIon in the constructor.

It will default to the zero value

Reply:
I tried this but goland complained about "Too few values".

Reply:
Did you specify the parameter names? 

for reference: https://github.com/amzn/ion-hash-go/pull/7/files#diff-fd1d7badfe852c7f98f4361f9138183fR35-R48

Reply:
Done.